c++/CMakeLists.txt

```diff
@@ -51,6 +46,10 @@ if(CAPNP_LITE AND BUILD_TOOLS)
   message(WARNING "Command-line tools will not be built with CAPNP_LITE.")
 endif()
 
+if(BUILD_TOOLS AND EXTERNAL_CAPNP)
+  message(SEND_ERROR "EXTERNAL_CAPNP cannot be set with BUILD_TOOLS.")
```

Comment:
> independent projects shouldn't be loaded together

This PR enables to include `capnp` with `add_subdirectory` instead of using `ExternalProject`. The `capnp` is bound into project as git submodule. I understand that it is not the most common scheme to compile capnp, but is there anything wrong about it? It works pretty nicely for me.